### PR TITLE
feat: Add ability to Link external helm app with devtron chart

### PR DESCRIPTION
**Pull request by manish-agrawal-ai**
Fix of #[1270](https://github.com/devtron-labs/devtron/issues/1270)

# Description
Functionality added for : 
1) Linking external helm chart with devtron APP

Fixes https://github.com/devtron-labs/devtron/issues/1270

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
By updating the helm App release with linking (API invoking)

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: pkg/appStore/deployment/AppStoreDeploymentService.go [hunk 1]
```diff
@@ -93,8 +103,15 @@ func (impl AppStoreDeploymentServiceImpl) AppStoreDeployOperationDB(installAppVe
 		return nil, err
 	}
 
+	var appInstallationMode string
```
Comment (nishant-d) [inline pkg/appStore/deployment/AppStoreDeploymentService.go:106]: we can directly assign the variable instead of if else

**Review by nishant-d — CHANGES_REQUESTED** [review]

File: api/appStore/deployment/AppStoreDeploymentRestHandler.go [hunk 2]
```diff
@@ -261,3 +267,55 @@ func (handler AppStoreDeploymentRestHandlerImpl) DeleteInstalledApp(w http.Respo
...
+	rbacObject := handler.enforcerUtilHelm.GetHelmObjectByClusterId(appIdentifier.ClusterId, appIdentifier.Namespace, appIdentifier.ReleaseName)
+	token := r.Header.Get("token")
+	if ok := handler.enforcer.Enforce(token, casbin.ResourceHelmApp, casbin.ActionUpdate, rbacObject); !ok {
+		common.WriteJsonResp(w, errors.New("unauthorized"), nil, http.StatusForbidden)
+		return
+	}
+	//RBAC enforcer Ends
+
+	// check if chart repo is active starts
+	isChartRepoActive, err := handler.appStoreDeploymentService.IsChartRepoActive(int(request.GetAppStoreApplicationVersionId()))
```
[the remark this answers is not in the document]
Comment (manish-agrawal-ai) [reply api/appStore/deployment/AppStoreDeploymentRestHandler.go:302]: done

File: api/appStore/deployment/AppStoreDeploymentRestHandler.go [hunk 3]
```diff
@@ -38,9 +41,10 @@ import (
 )
 
 type AppStoreDeploymentRestHandler interface {
-	CreateInstalledApp(w http.ResponseWriter, r *http.Request)
+	InstallApp(w http.ResponseWriter, r *http.Request)
 	GetInstalledAppsByAppStoreId(w http.ResponseWriter, r *http.Request)
 	DeleteInstalledApp(w http.ResponseWriter, r *http.Request)
+	UpdateHelmApplicationWithChartStoreLinking(w http.ResponseWriter, r *http.Request)
```
[the remark this answers is not in the document]
Comment (manish-agrawal-ai) [reply api/appStore/deployment/AppStoreDeploymentRestHandler.go:47]: name refactored. 
For already app, we are not calling the same method. currently, update is happening externally to helm directly. we have to update link also. this item we have in our pipeline.

File: pkg/appStore/deployment/AppStoreDeploymentService.go (hunk 1 above)
Comment (manish-agrawal-ai) [reply pkg/appStore/deployment/AppStoreDeploymentService.go:106]: if else is needed due to installAppVersionRequest.AppOfferingMode check in if condition

**Review by nishant-d — APPROVED** [review]

**Pull request merged**